SimTKcommon/include/SimTKcommon/internal/ParallelExecutor.h

```diff
@@ -80,13 +91,21 @@ class SimTK_SimTKCOMMON_EXPORT ParallelExecutor : public PIMPLHandle<ParallelExe
      */
     void execute(Task& task, int times);
     /**
-     * Get the number of available processor cores.
+     * Get the total number of available processor cores (physical cores and
+     * hyperthreads on Intel architecture). If the number of threads is not
+     * computable or well defined, the function returns 0.
      */
     static int getNumProcessors();
     /**
-     * Determine whether the thread invoking this method is a worker thread created by ParallelExecutor.
+     * Determine whether the thread invoking this method is a worker thread
+     * created by ParallelExecutor.
      */
     static bool isWorkerThread();
+    /**
+     * Get the maximum number of thread contexts that the ParallelExecutor is
+     * currently allowed to use.
+     */
+    int getNumMaxThreads() const;
```

Comment:
What about `getMaxThreads`? That's what Thomas named the OpenSim environment variable (`OPENSIM_CMC_MAX_THREADS`). OpenMP has a method `omp_get_max_threads()` (and also a separate `omp_get_num_threads()`). http://www.plutospin.com/files/OpenMP_reference.pdf